test/testdata/lsp/code_actions/extract_class_func.rb

```diff
@@ -0,0 +1,36 @@
+# typed: strict
+# selective-apply-code-action: refactor.extract
+
+module Foo
+  extend T::Sig
+
+  sig {returns(String)}
+  def self.greeting
+         # ^^^^^^^^ apply-code-action: [A] Extract method to module
```

Comment:
It looks like there are no tests for what happens if the user attempts to use this code action on an instance method. Can we get a test for that?

Reply:
Another test: what happens if the user wrote `private_class_method def foo` instead of just `def foo`?